[ci] Optimize ESLint performance with caching and generated file excl…

christian-byrne opened the pull request:
…usion

## Summary

Optimize eslint with caching and ignoring generated type files.

## Changes

1. Ignore Generated Files: Two large generated files identified: comfyRegistryTypes.ts (550KB) and generatedManagerTypes.ts (41KB). Currently not ignored in ESLint config despite being marked as linguist-generated in .gitattributes. Reduces file count by only 2/771 but removes 591KB of content.
- Also change github workflows in which the types are generated to run lint without caching since these are the only time the types are modified.                                                                                                            3. Enable ESLint Caching to only run on changed files                                                                  

## Review Focus

Resolves https://github.com/Comfy-Org/ComfyUI_frontend/issues/4896.

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-4926-ci-Optimize-ESLint-performance-with-caching-and-generated-file-excl-24c6d73d3650815a81e9fd2e77f6ac16) by [Unito](https://www.unito.io)

christian-byrne committed:
[ci] Optimize ESLint performance with caching and generated file exclusion

Co-authored-by: Qwen-Coder <qwen-coder@alibabacloud.com>

christian-byrne committed:
ci: Add linting step for generated API types

- Add ESLint validation to registry and manager type generation workflows
- Ensures generated files meet code standards even though they're ignored in regular lint runs
- Uses --no-ignore flag to lint specific files despite ESLint ignore patterns

Co-authored-by: Qwen-Coder <qwen-coder@alibabacloud.com>

benceruleanlu commented on package.json:
```diff
@@ -21,8 +21,8 @@
     "preview": "vite preview",
-    "lint": "eslint src",
-    "lint:fix": "eslint src --fix",
+    "lint": "eslint src --cache",
+    "lint:fix": "eslint src --cache --fix",
```
WDYT about having a new npm script for linting with no cache? not important though

christian-byrne replied:
Good idea

benceruleanlu reviewed (dismissed):
also, I think we need `.eslintcache` to be added to `.gitignore`

christian-byrne committed:
ci: Add auto-fix linting step for generated API types

- Add ESLint validation with auto-fix to registry and manager type generation workflows
- Ensures generated files meet code standards and automatically fixes any issues
- Uses --no-ignore --fix flags to lint and fix specific files despite ESLint ignore patterns

Co-authored-by: Qwen-Coder <qwen-coder@alibabacloud.com>

christian-byrne committed:
[ci] Add lint:no-cache npm script and update workflows

- Add lint:no-cache script to package.json for linting without cache
- Update workflow files to use npm script instead of npx eslint --no-ignore
- Add .eslintcache to .gitignore
